floris/simulation/turbine.py

```diff
@@ -319,6 +489,38 @@ def __attrs_post_init__(self) -> None:
         object.__setattr__(self, "wind_speed", self.wind_speed[duplicate_filter])
 
 
+@define
+class TiltTable(FromDictMixin):
+    """Helper class to convert the dictionary and list-based inputs to a object of arrays.
+
+    Args:
+        tilt (NDArrayFloat): The tilt angle at a given windspeed.
+        wind_speeds (NDArrayFloat): Windspeed values, m/s.
+
+    Raises:
+        ValueError: Raised if tilt and wind_speeds are not all 1-d array-like shapes.
+        ValueError: Raised if tilt and wind_speeds don't have the same number of values.
+    """
+    tilt: NDArrayFloat = field(converter=floris_array_converter)
+    wind_speeds: NDArrayFloat = field(converter=floris_array_converter)
+
+    def __attrs_post_init__(self) -> None:
+        # Validate the power, thrust, and wind speed inputs.
+
+        inputs = (self.tilt, self.wind_speeds)
+
+        if any(el.ndim > 1 for el in inputs):
+            raise ValueError("tilt and wind_speed inputs must be 1-D.")
+
+        if len({self.tilt.size, self.wind_speeds.size}) > 1:
+            raise ValueError("tilt and wind_speed tables must be the same size.")
+
+        # Remove any duplicate wind speed entries
+        _, duplicate_filter = np.unique(self.wind_speeds, return_index=True)
+        object.__setattr__(self, "tilt", self.tilt[duplicate_filter])
```

Comment:
Just curious about these two lines. What's different between this and the following;
```python
self.tilt = self.tilt[duplicate_filter]
```

Reply:
After talking with @RHammond2, my understanding is that this method was used when these tables were initially immutable (I modeled this after the pattern in PowerThrustTable). But sounds like the change was made along the way to have PowerThrustTable to be mutable, so I've updated to your suggestion in both tables.